tests/storage-vm

```diff
@@ -45,6 +45,12 @@ for poolDriver in $poolDriverList; do
                 [ "$(lxc storage get "${poolName}" size)" = "20GiB" ]
         fi
 
+        # Ensure non-power-of-two sizes are rounded appropriately for zfs
+        if [ "${poolDriver}" = "zfs" ]; then
```

Comment:
I would have expected:
```
        if [ "${poolDriver}" != "powerflex" ]; then
```
But maybe the rounding logic is really `zfs` specific? I thought something similar was done for all of them at the very least to be 4k-aligned (or 1MiB-aligned) and avoid the `read-write-modify` pathological case.

Reply:
`MinBlockBoundary` is [here](https://github.com/canonical/lxd/blob/main/lxd/storage/drivers/utils.go#L25); the code is a bit convoluted but it should be used for dir, btrfs, zfs, and lvm. I don't see any uses in ceph or powerflex. Happy to make the change if you'd like.

Reply:
After thinking some more, having the outer raw file/zvol/etc not be aligned to 4KiB or 1MiB shouldn't matter as the inner OS of the VM will likely have logic to do the alignment itself anyway.

Let's keep the test as is (`zfs` only) as creating a 13GB file on dir is probably too slow anyway. Thanks for digging.